gems/smithy/lib/smithy/welds/endpoints.rb

```diff
@@ -0,0 +1,134 @@
+# frozen_string_literal: true
+
+module Smithy
+  module Welds
+    # Provides default endpoint builtin/function bindings.
+    class Endpoints < Weld
+      def preprocess(model)
+        model['shapes'].select { |_k, s| s['type'] == 'service' }.each_value do |shape|
+          add_default_endpoints(shape['traits']) unless shape['traits']['smithy.rules#endpointRuleSet']
+        end
+      end
+
+      def built_in_bindings
+        [Vise::Endpoints::BuiltInBinding.new(
+          id: 'SDK::Endpoint',
+          render_config: proc do |_plan|
+            <<-ADD_OPTION
+      option(
+        :endpoint,
+        doc_type: String,
+        docstring: "Custom Endpoint"
+      )
+            ADD_OPTION
+          end,
+          render_build: proc do |_plan, _operation|
+            'config.endpoint'
+          end,
+          render_test_set: proc do |_plan, _operation, _node|
+          end
+        )]
+      end
+
+      # rubocop:disable Metrics/MethodLength
+      def function_bindings
+        [
+          Vise::Endpoints::FunctionBinding.new(
```

Comment:
Similar to the vise comments before, I think these should really just be a collection of hashes/lists from the model (it could really just be the same format as the model, too), and then passed to the view, which may wrap those to help with rendering.

Reply:
I'd disagree since these are the public extension point - I think having a class, even if its a simple one, help define that interface.

Reply:
Maybe these should be in Smithy::Endpoints and not vise. Vise doesn't seem correct. But I also don't think a class is necessary.